File: beginner/pagina12/3089/3089.ts
const fs = require("fs")
const input = fs.readFileSync("/dev/std", "utf8")

while(true){

    const n = Number(input.shift())

    if(n === 0 ) break

    const presentes = input.shift().trim().split(' ').map(Number)

    let pares = []
    for(let i = 0; i < n; i++){
        pares.push(presentes[i] + presentes[2 * n - i - 1])
    }

    console.log(`${Math.max(...pares)} ${Math.min(...pares)}`)
}

File: beginner/pagina12/3089/3089.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){

    while(true){

        vector<int> presentes;
        vector<int> pares;
        int n;

        cin >> n;
        
        if(n == 0){
            break;
        }

        for(int i = 0; i < 2 * n; i++){
            int curr;
            cin >> curr;
            presentes.push_back(curr);
        }

        for(int i = 0; i < n; i++){
            pares.push_back(presentes[i] + presentes[2 * n - i - 1]);
        }

        int max = pares[0];
        int min = pares[0];

        for(int i = 1; i < n; i++){
            if(pares[i] > max)
                max = pares[i];
            if(pares[i] < min)
                min = pares[i];
        }

        cout << max << " " << min << endl;

    }
    return 0;
}